runtime/runtime/src/deterministic_account_id.rs

```diff
@@ -13,90 +13,49 @@ use near_primitives_core::deterministic_account_id::DeterministicAccountStateIni
 use near_store::{StorageError, TrieUpdate};
 use near_vm_runner::logic::ProtocolVersion;
 
-/// State machine of a deterministic account id on chain.
-///
-/// This follows the definition in [NEP-616](https://github.com/near/NEPs/pull/616/).
-enum AccountState {
-    /// There were no accepted receipts on this account, so it doesn't have any
-    /// data (or the contract was deleted). Initially all deterministic accounts
-    /// are in this state.
-    NonExisting,
-    /// Account exists and contains balance and meta info but no contract has
-    /// been deployed. An account enters this state, for example, when it was in
-    /// a non-exist state, and another account sent native transfer to it.
-    Uninit,
-    /// Account has contract code deployed, persistent data and balance. An
-    /// account enters this state when it was in non-exist or uninit state and
-    /// there was a valid incoming StateInit action.
-    Active,
-}
-
-impl AccountState {
-    fn of(maybe_account: Option<&Account>) -> AccountState {
-        let Some(account) = maybe_account else {
-            return AccountState::NonExisting;
-        };
-
-        if account.contract().is_none() {
-            return AccountState::Uninit;
-        }
-
-        AccountState::Active
-    }
-}
-
 pub(crate) fn action_deterministic_state_init(
     state_update: &mut TrieUpdate,
     apply_state: &ApplyState,
-    account: &mut Option<Account>,
+    maybe_account: &mut Option<Account>,
```

Comment:
unrelated to your PR so just ranting - I don't like mutable parameters, they are the worst

Reply:
yes, 100%, I've just had a chat with @graphite and we agreed that runtime code should be refactored to get rid of this pattern